### ISSUE TITLE
Use prepare_jinja2_environment hook instead

### ISSUE THREAD

**Issue by simonw** [simonw/datasette-edit-templates#1]
I removed the `load_template` hook in https://github.com/simonw/datasette/issues/1073 so this prototype plugin no longer works.

It should be possible to build it against `prepare_jinja2_environment` instead.

**Comment by simonw** [comment]
Assuming `env.loader` is already a `ChoiceLoader`, this is a case of injecting a new `FunctionLoader(load_template)` at the start of that list.

```
def load_template(name):
    uptodatefunc = lambda: True
    source = load_from_db(name)
    if source is None:
        return None
    return source, name, uptodatefunc
```

**Comment by simonw** [comment]
The trick here is that `FunctionLoader` cannot make async calls - see https://github.com/pallets/jinja/issues/1304

So... I'm going to load all templates from the database into an in-memory dictionary and read them from there. Then the code that edits the templates in the DB can also update that in-memory dictionary.

**Comment by simonw** [comment]
Released as `0.1a0`.

**Comment by simonw** [comment]
Demo:

![edit-templates](https://user-images.githubusercontent.com/9599/190263101-1f70f715-6eb2-4ea4-b791-1194dabda640.gif)
